packages/editor/src/plugins/fill-in-the-blanks-exercise/editor.tsx

```diff
@@ -26,27 +37,66 @@ export function FillInTheBlanksExerciseEditor(
   })
 
   if (!textPluginState) return null
+  console.log('Editor: ', { isEditMode, textPluginState, props })
 
   return (
     <div className="mb-12 mt-10 pt-4">
-      {hasFocus ? (
+      {hasFocus || !isEditMode ? (
         // TODO: Add button to toggle between fill-in-the-blanks and drag-and-drop
         <PluginToolbar
           pluginType={EditorPluginType.FillInTheBlanksExercise}
           className="!left-[21px] top-[-33px] w-[calc(100%-37px)]"
           pluginControls={<InteractiveToolbarTools id={props.id} />}
+          pluginSettings={
+            <>
+              <button
+                onClick={toggleRenderEditMode}
+                className="mr-2 rounded-md border border-gray-500 px-1 text-sm transition-all hover:bg-editor-primary-200 focus-visible:bg-editor-primary-200"
+                data-qa="plugin-fill-in-the-blanks-toggle"
+              >
+                {isEditMode ? (
+                  <>
+                    <FaIcon icon={faEye} /> {blanksExerciseStrings.preview}
+                  </>
+                ) : (
+                  <>
+                    <FaIcon icon={faPen} /> {blanksExerciseStrings.edit}
+                  </>
+                )}
+              </button>
+            </>
```

Comment:
all fragments here can be removed

Reply:
while this is probably a bit clearer I stil prefer the approach [here](https://github.com/serlo/frontend/blob/91c884a791814cf9329176d99c2c235d92ae79bd/packages/editor/src/plugins/input-exercise/toolbar.tsx#L31-L45) since it does not cause the button to suddenly change length.